### PR TITLE
Bump go-sdk version to 0.36.0

### PR DESCRIPTION
**Pull request by edwardfeng-db**
## Changes

- Bump up go-sdk version to `0.36.0`

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [ ] using Go SDK

### CODE REVIEW

**Review by hectorcast-db — APPROVED** [review]

**Review by alexott — CHANGES_REQUESTED** [review]
We already had PR to bump versions, but we need to make code changes to accommodate it: https://github.com/databricks/terraform-provider-databricks/pull/3389

File: catalog/data_catalogs.go [hunk 1]
```diff
@@ -4,14 +4,16 @@ import (
...
+	"github.com/databricks/databricks-sdk-go/service/catalog"
 	"github.com/databricks/terraform-provider-databricks/common"
 )
 
 func DataSourceCatalogs() common.Resource {
 	return common.WorkspaceData(func(ctx context.Context, data *struct {
 		Ids []string `json:"ids,omitempty" tf:"computed,slice_set"`
 	}, w *databricks.WorkspaceClient) error {
-		catalogs, err := w.Catalogs.ListAll(ctx)
+		var req catalog.ListCatalogsRequest = catalog.ListCatalogsRequest{}
```
Comment (alexott) [inline catalog/data_catalogs.go:15]: why we need to have variable here, why not simple `ListAll(ctx, catalog.ListCatalogsRequest{})` ?

File: catalog/data_catalogs_test.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ func TestCatalogsData(t *testing.T) {
 		Fixtures: []qa.HTTPFixture{
 			{
 				Method:   "GET",
-				Resource: "/api/2.1/unity-catalog/catalogs",
+				Resource: "/api/2.1/unity-catalog/catalogs?",
```
Comment (alexott) [inline catalog/data_catalogs_test.go:15]: Can we fix SDK to not produce `?` when there is no parameters?

File: catalog/resource_storage_credential.go [hunk 3]
```diff
@@ -130,11 +130,6 @@ func ResourceStorageCredential() common.Resource {
 			common.DataToStructPointer(d, storageCredentialSchema, &update)
 			update.Name = d.Id()
 			update.Force = force
-			// We need to set them to empty since these fields are computed
-			if _, ok := d.GetOk("aws_iam_role"); ok {
-				update.AwsIamRole.UnityCatalogIamArn = ""
-				update.AwsIamRole.ExternalId = ""
-			}
```
Comment (alexott) [inline catalog/resource_storage_credential.go:137]: Hmmm, @nkvuong - what do you think, can we remove this?

**Review by alexott — CHANGES_REQUESTED** [review]
We still need to update docs - delete references to removed attributes

File: catalog/resource_storage_credential.go (hunk 3 above)
Comment (edwardfeng-db) [reply catalog/resource_storage_credential.go:137]: It seems like these two are already marked as `output only` in the current doc, do we still need to make any change?

Comment (alexott) [reply catalog/resource_storage_credential.go:137]: we need to remove them from the docs.

Comment (edwardfeng-db) [reply catalog/resource_storage_credential.go:137]: Removed, PTAL

**Review by alexott — APPROVED** [review]

**Pull request merged**